File: XuZeng/739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& T) {
        int n=T.size();
        vector<int> res(n,0);
        res[n-1]=0;
        for(int i=n-2;i>=0;i--){
            int temp=i+1;
            while(1){
                if(T[i]>=T[temp]){
                    if(res[temp]==0){
                        res[i]=0;
                        break;
                    }
                    else
                        temp+=res[temp];
                }
                else{
                    res[i]=temp-i;
                    break;
                }
            }
        }
        return res;
    }
};

File: sparrowsong/739.go
package leetcode

func dailyTemperatures(T []int) []int {
	res := make([]int, len(T))
	helper := []int{}
	for idx := len(T) - 1; idx >= 0; idx-- {
		res[idx] = 0
		for i := len(helper) - 1; i >= 0; i-- {
			if helper[i] > T[idx] {
				res[idx] = len(helper) - i
				break
			}
		}
		helper = append(helper, T[idx])
	}
	return res
}
